### PR TITLE
models: CLIP zero-shot image classification (both encoders on the ANE) (#179)

### LINKED ISSUE #179: examples: CLIP zero-shot image classification (both encoders on the ANE)
A high-visibility demo: **CLIP zero-shot image classification** end to end on the ANE -- feed an image plus a list of candidate text labels, run both encoders on the engine, and print the softmax over label similarities. Shareable and immediately legible, in the spirit of the `examples/vit.py` / `examples/sentence_embeddings.py` demos.

**Why this is more involved than the other example issues** (hence `help wanted`): it needs a real dual-encoder model. The vision tower is a ViT (see `examples/vit.py` and the native attention path), the text tower is a small transformer encoder (see `examples/sentence_transformers_ane.py` / `sentence_embeddings.py`), and the two projections meet in a shared embedding space. Loading pretrained CLIP weights (e.g. an open CLIP checkpoint) and mapping them onto the aneforge graph is the bulk of the work.

**Suggested scope:**
- `examples/clip_zero_shot.py`: load a small CLIP checkpoint, encode one image + N candidate captions, print the ranked scores.
- Reuse the existing ViT and text-encoder building blocks rather than writing new ops.
- Keep weights out of the repo (download or point at `~/Models`), consistent with the other model demos.

**Testing:** compare the top-1 label and the similarity scores against the reference PyTorch CLIP on a couple of images. Requires an Apple Silicon Mac. Ping on the issue if you pick it up -- happy to help with the weight mapping.

### PR DESCRIPTION
**Pull request by axiom-of-choice**
READY:

Resolves #179

This PR implements full end-to-end **CLIP zero-shot image classification** on the Apple Neural Engine, running both the Vision Transformer () and causal Text Transformer () on the ANE.

### Architecture & Implementation
- **Vision Tower**: 
  - Patch embedding via `space_to_depth(P)` + 1x1 conv (strided 32x32 conv avoided for ANE compatibility).
  - Pre-norm Transformer stack with QuickGELU (`x * sigmoid(1.702 * x)`).
  - `[CLS]` token pooling + linear visual projection + in-graph L2 normalization.
- **Text Tower**:
  - Token and learned position embedding gather.
  - Causal self-attention with precomputed triangular mask + QuickGELU MLP layers.
  - `[EOS]` / EOT token state extraction + text projection + in-graph L2 normalization.
- **Unified API**:
  - `af.load_clip(name)` / `CLIP` class with `.encode_image()`, `.encode_text()`, and `.classify(image, candidate_labels)`.
- **Demo & Documentation**:
  - `examples/clip_zero_shot.py`: end-to-end zero-shot classification example scoring candidate labels against an image with probability ranking and benchmarks vs Hugging Face PyTorch fp32.
  - Documented in `docs/developer/models.md`.

### On-Device Verification (M2 Pro, A15 ANE)
- `examples/clip_zero_shot.py`:
  - **Vision embedding cosine vs HF fp32**: `0.99924` (PASS)
  - **Text embeddings min cosine vs HF fp32**: `0.99984` (PASS)
  - **Vision latency on ANE**: `8.51 ms`
  - **Text latency on ANE**: `2.26 ms/prompt`
  - **Total classify latency (5 candidate labels)**: `12.85 ms`
  - **Top-1 classification match**: Exact match with Hugging Face probabilities (`diff < 1%`).
- **Tests**: `tests/test_clip.py` (4/4 PASS).
- **Corpus Gate**: `tests/run_corpus.py` (90/90 PASS, `GATE: GREEN`).

Code matches house style (2-space, ASCII, clean pyright/ruff, 10/10 pylint).

### CODE REVIEW

File: aneforge/models.py
Comment (axiom-of-choice) [inline aneforge/models.py:1]: Might conflict with https://github.com/sbryngelson/ANEForge/pull/224

File: aneforge/models.py [hunk 1]
```diff
@@ -559,6 +559,155 @@ def release(self) -> None:
...
+class CLIP:
+  """CLIP dual-encoder model from Hugging Face (CLIPModel), running both vision and text encoders on the ANE.
+  `.encode_image(image)` -> [1, proj_dim] normalized image embedding.
+  `.encode_text(texts)` -> [N, proj_dim] normalized text embeddings.
+  `.classify(image, candidate_labels)` -> sorted list of (label, probability)."""
+
+  def __init__(self, name: str = "openai/clip-vit-base-patch32", int8: bool = False) -> None:
+    from transformers import AutoConfig, AutoTokenizer, CLIPModel  # lazy
+    cfg = AutoConfig.from_pretrained(name)
+    if getattr(cfg, "model_type", None) != "clip":
```
Comment (axiom-of-choice) [inline aneforge/models.py:557]: After https://github.com/sbryngelson/ANEForge/pull/224 is merged, into would take Enum approach there.

Comment (axiom-of-choice) [reply aneforge/models.py:557]: TO-DO (deferred) as there's only one "model_type" here being matched. Whem this grows, we will need to create an ENUM.

**Comment by sbryngelson** [comment]
Tested end-to-end on an M5 Pro. examples/clip_zero_shot.py: RESULT PASS -- vision embedding cosine 0.99923 vs HF, text embeddings min cosine 0.99985, top-1 classification matches HF with the same ranking. Both towers run on the ANE (vision 4.1 ms, text 1.4 ms/prompt, total classify 8.9 ms). Off-device tests 5/5. The QuickGELU + EOS pooling + in-graph L2 normalization all validate against HF fp32. Nice -- both encoders of CLIP on the engine closes #179. Merging.
